### PR TITLE
[Master] Add new options in xml conversion service

### PR DESCRIPTION
**Pull request by mindula**
## Purpose
$title

Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by prakanth97** [comment]
There is a windows build failure, Shall we fix that one

**Review by hasithaa — DISMISSED** [review]

**Review by hasithaa — APPROVED** [review]

File: misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordRequest.java [hunk 1]
```diff
@@ -25,53 +25,39 @@
...
-    private String textFieldName;
-    private boolean withNameSpace;
+    private final String xmlValue;
+    private final boolean isRecordTypeDesc;
+    private final boolean isClosed;
+    private final boolean forceFormatRecordFields;
+    private final String textFieldName;
+    private final boolean withNameSpace;
 
     public XMLToRecordRequest(String xmlValue, boolean isRecordTypeDesc, boolean isClosed,
```
Comment (prakanth97) [inline misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordRequest.java:35]: Is this exposed to the tooling side? if so, modifying the constructor may cause error?

Comment (LakshanWeerasinghe) [reply misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordRequest.java:35]: They are sending jsonrpc request from the vscode plugin. So it won't be a problem.

Comment (prakanth97) [reply misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordRequest.java:35]: Tooling team will change the request accordingly.

**Review by prakanth97 — APPROVED** [review]
LGTM

**Pull request merged**